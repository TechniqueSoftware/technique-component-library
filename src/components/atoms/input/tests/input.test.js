import React from 'react'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import Input from '..'

const firstNameProps = {
  name: 'firstName',
  type: 'text',
  label: 'First Name',
  value: '',
  error: '',
  required: true,
  relatedField: null,
  validated: false,
  onChange: jest.fn(),
  onBlur: jest.fn()
}
const lastNameProps = {
  name: 'LastName',
  type: 'text',
  label: 'Last Name',
  value: '',
  error: '',
  required: true,
  relatedField: null,
  validated: false,
  onChange: jest.fn(),
  onBlur: jest.fn()
}

describe('<Input/>', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(<Input {...firstNameProps} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render', () => {
    render(<Input {...firstNameProps} />)
    expect(screen.getByLabelText('First Name *')).toBeInTheDocument()
  })

  it('should render horizontal', () => {
    render(<Input {...firstNameProps} layoutHorizontal />)

    expect(screen.getByLabelText('First Name *')).toBeInTheDocument()
    expect(screen.getByLabelText('First Name *').parentElement).toHaveClass('col')
  })

  it('should show that the onChange function updates the props value', () => {
    render(<Input {...firstNameProps} />)
    userEvent.type(screen.getByLabelText('First Name *'), 't')
    expect(firstNameProps.onChange).toHaveBeenCalled()
  })

  it('should validate on exit ', () => {
    render(<><Input {...firstNameProps} /> <Input {...lastNameProps} /></>)
    userEvent.type(screen.getByLabelText('First Name *'), 'ts')
    expect(firstNameProps.onChange).toHaveBeenCalled()
    userEvent.tab()
    expect(screen.getByLabelText('Last Name *')).toHaveFocus()
  })
})
