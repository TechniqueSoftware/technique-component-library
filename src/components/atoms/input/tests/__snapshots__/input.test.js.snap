// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input/> should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="mb-3"
  >
    <label
      class="form-label"
      for="firstName"
    >
      First Name *
    </label>
    <input
      class="form-control"
      id="firstName"
      name="firstName"
      required=""
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;
